feat: update progress bar default fill to horizontal gradient

camsimony opened the pull request:
# Description

This PR updates the default progress bar fill color from solid red (#FF3E14) to a horizontal gradient (#FDA4AF → #FF3E14) in the Pine design system, as specified in DSS-1497.

## Changes Made

- **Added CSS variables** for gradient colors (`--progress-gradient-start` and `--progress-gradient-end`) to enable future customization
- **Updated progress pseudo-elements** (`::-webkit-progress-value` and `::-moz-progress-bar`) to use `background` with gradient fallback instead of `background-color`
- **Preserved existing CSS variable system** to ensure custom colors still override the default gradient

The implementation uses CSS variable fallback logic: `var(--color-progress-fill, linear-gradient(to right, var(--progress-gradient-start), var(--progress-gradient-end)))` to maintain backward compatibility.

Fixes DSS-1497

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

⚠️ **Important**: Visual verification is incomplete and requires human review.

- [x] Compiled successfully with no build errors
- [x] Pre-commit hooks passed (lint, validate-commit-msg)
- [x] Tested manually in Storybook at 50% progress
- [ ] **Visual gradient verification needed** - Could not confirm gradient rendering in browser testing
- [ ] Custom color override testing
- [ ] Cross-browser compatibility testing
- [ ] RTL layout behavior testing

**Test Configuration**:
- Pine versions: Local development build
- OS: Ubuntu Linux
- Browsers: Chrome (Storybook testing)
- Tested component: Progress component in Pine Storybook

# Critical Review Areas

Please pay special attention to:

1. **Visual appearance**: Verify the gradient (#FDA4AF → #FF3E14) renders correctly
2. **Custom color preservation**: Test that setting `--color-progress-fill` still overrides the gradient
3. **Cross-browser compatibility**: Test in Chrome, Firefox, and Safari (webkit vs moz pseudo-elements)
4. **RTL layout behavior**: Gradient direction may need adjustment for right-to-left layouts
5. **CSS fallback logic**: Ensure `var(--color-progress-fill, linear-gradient(...))` behaves as expected

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

---

**Link to Devin run**: https://app.devin.ai/sessions/f76dcde5eca0423a9fcdba1671311c27  
**Requested by**: Cameron Simony (@cameronsimony)

**Note**: This PR is part of a coordinated update across both Pine and Sage design systems. A corresponding PR for Sage will be created separately.

devin-ai-integration[bot] committed:
feat: update progress bar default fill to horizontal gradient

- Replace solid red (#FF3E14) with gradient (#FDA4AF → #FF3E14)
- Add gradient colors as CSS variables for future flexibility
- Preserve existing CSS variable fallback system for custom colors
- Only affects default variant, custom colors remain unchanged

DSS-1497

Co-Authored-By: Cameron Simony <cameron.simony@kajabi.com>
Co-Authored-By: cameron.simony@kajabi.com <cameron.simony@gmail.com>

devin-ai-integration[bot] committed:
fix: update CSS selector to only override gradient when fill-color has value

- Change :host([fill-color]) to :host([fill-color]:not([fill-color=""]))
- This prevents empty fill-color attribute from overriding the gradient
- Ensures default progress bar shows gradient while custom colors still work

Co-Authored-By: cameron.simony@kajabi.com <cameron.simony@gmail.com>

devin-ai-integration[bot] committed:
feat: replace hex values with Pine design tokens for gradient

- Use --pine-color-red-300 for gradient start (lighter color)
- Use --pine-color-brand for gradient end (brand color)
- Addresses GitHub comment about using design tokens instead of hex values

Co-Authored-By: cameron.simony@kajabi.com <cameron.simony@gmail.com>

QuintonJason reviewed (changes requested):
aside

@cameronsimony Rejecting since a change is occurring that isn't represented in the corresponding mercury component design file: https://www.figma.com/design/CC1YmaGKHnsvB28yLY9mEH/%E2%9D%96-Mercury-components?m=auto&node-id=737-22323

cameronsimony commented:
Updated the figma side [here](https://www.figma.com/design/CC1YmaGKHnsvB28yLY9mEH/%E2%9D%96-Mercury-components?node-id=823-30602&t=CxAqcZ2qRQYk7aYm-11)

I'll communicate to the teams before a merge 👍

pixelflips reviewed (commented):
aside. ignore last comment

pixelflips commented on libs/core/src/components/pds-progress/pds-progress.scss:
```diff
@@ -44,15 +45,23 @@ progress::-webkit-progress-bar {
   border-radius: var(--pine-dimension-2xs);
 }
 
+:host([fill-color]:not([fill-color=""])) progress::-webkit-progress-value {
+  background: var(--color-progress-fill);
+}
+
+:host([fill-color]:not([fill-color=""])) progress::-moz-progress-bar {
+  background: var(--color-progress-fill);
+}
```
Could these be comma-separated and combined into one?

devin-ai-integration[bot] committed:
refactor: combine webkit and moz progress bar selectors

- Consolidate custom color override selectors into comma-separated rule
- Improves code maintainability while preserving functionality
- Addresses GitHub comment from pixelflips about combining selectors

Jira: DSS-1497
Co-Authored-By: cameron.simony@kajabi.com <cameron.simony@gmail.com>

devin-ai-integration[bot] committed:
fix: update gradient colors to red-300 → mercury-500

- Change gradient end color from brand to mercury-500
- Addresses feedback about incorrect gradient color specification
- Maintains existing functionality with correct design tokens

Jira: DSS-1497
Co-Authored-By: cameron.simony@kajabi.com <cameron.simony@gmail.com>

devin-ai-integration[bot] committed:
fix: use var(--pine-color-brand) for gradient end color

- Address GitHub comment from pixelflips to use brand token
- More semantically correct than direct mercury-500 reference
- Maintains same visual appearance while using proper token

Jira: DSS-1497
Co-Authored-By: cameron.simony@kajabi.com <cameron.simony@gmail.com>

pixelflips reviewed (approved):
LGTM! Please do not merge until @QuintonJason approves as well.

QuintonJason reviewed (changes requested):
@cameronsimony @pixelflips Requesting changes because this fix feels too narrowly scoped. Since this introduces a gradient token into the system, we should take a system-wide approach to defining and using gradient components. The progress bar should consume this shared token rather than introducing a one-off implementation.

We’ll also need to build out the gradient options in the design files so we can explore and align on all possible variations before locking in an implementation.

Additionally, the current code does not allow for a custom gradient, which is a limitation we should address to maintain flexibility.

pixelflips commented:
aside

@QuintonJason  i agree. any broader gradient implementation will need a comprehensive plan. for this specific change though, we could move forward with a quick follow-up to add a component-level token, but that depends on the scope of what we're trying to achieve.

@cameronsimony could you clarify the intended scope of these changes? if this is solely updating the default progress bar styling, we can proceed and follow up immediately with the component token. however, if the plan includes multiple variants or custom gradient support, that would require a more extensive project approach.

quinton raises some important considerations, so we'd appreciate clarification on the design goals before merging.

pixelflips commented:
@cameronsimony have you had a chance to review the comments above?
